pkg/common/helpers.go

```diff
@@ -49,3 +67,89 @@ func IsExpired(startTime int64, ttlTime int64) bool {
 	}
 	return false
 }
+
+// NewWFClient -- declare new workflow client
+func NewWFClient(cfg *rest.Config) wfv1versioned.Interface {
+	cli, err := wfv1versioned.NewForConfig(cfg)
+	if err != nil {
+		panic(err)
+	}
+	return cli
+}
+
+// NewAddonClient - declare new addon client
+func NewAddonClient(cfg *rest.Config) addonv1versioned.Interface {
+	cli, err := addonv1versioned.NewForConfig(cfg)
+	if err != nil {
+		panic(err)
+	}
+	return cli
+}
+
+// return cluster config
+func InClusterConfig() (*rest.Config, error) {
+	k8shost := "KUBERNETES_SERVICE_HOST"
+	k8sport := "KUBERNETES_SERVICE_PORT"
+	if len(os.Getenv(k8shost)) == 0 {
+		addrs, err := net.LookupHost("kubernetes.default.svc")
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+		os.Setenv(k8shost, addrs[0])
+	}
+	if len(os.Getenv(k8sport)) == 0 {
+		os.Setenv(k8sport, "443")
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+// NewK8sClient defines kubernetes client
+func NewK8sClient(kubeconfigPath string) (kubernetes.Interface, error) {
+	if kubeconfigPath == "" {
+		return nil, fmt.Errorf("kubeconfig should be configured a valid value")
+	}
+	var kubeconfig *rest.Config
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load kubeconfig from %s: %v", kubeconfigPath, err)
+	}
+	kubeconfig = config
+
+	client, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create a client: %v", err)
+	}
+	return client, nil
+}
+
+func WorkFlowFromUnstructured(un *unstructured.Unstructured) (*wfv1.Workflow, error) {
+	var wf wfv1.Workflow
+	err := FromUnstructuredObj(un, &wf)
+	return &wf, err
+}
+
+func FromUnstructured(un *unstructured.Unstructured) (*addonv1.Addon, error) {
+	var addon addonv1.Addon
+	err := FromUnstructuredObj(un, &addon)
+	return &addon, err
+}
+
+func FromUnstructuredObj(un *unstructured.Unstructured, v interface{}) error {
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, v)
+	if err != nil {
+		if err.Error() == "cannot convert int64 to v1alpha1.AnyString" {
```

Comment:
Why do we need this specific error check?

Reply:
Yea, this is error converting int64 string to anystring has this workaround.

Reply:
db6f893de3812e442ece0edc1be00e9732a6902d

Reply:
I don't believe this is the correct way to handle this, there are conversion functions that can registered with a scheme that should handle this case.

See example https://github.com/kubernetes/apimachinery/blob/release-1.21/pkg/runtime/conversion.go#L175-L179

Reply:
In our case, the register already done within informer, here is edge case handlers.